config/v1/types_infrastructure.go

```diff
@@ -1370,6 +1373,89 @@ type NutanixPlatformSpec struct {
 	// +listType=map
 	// +listMapKey=name
 	PrismElements []NutanixPrismElementEndpoint `json:"prismElements"`
+
+	// failureDomains configures failure domains information for the Nutanix platform.
+	// If this is omitted, failure domains will not be used to spread Machines.
+	// +optional
+	FailureDomains []NutanixFailureDomain `json:"failureDomains"`
+}
+
+// NutanixFailureDomain configures failure domain information for the Nutanix platform.
+type NutanixFailureDomain struct {
+	// name defines the unique name of a failure domain.
```

Comment:
The comment should be expanded to explain the maximum/minimum length requirements for this field.

Do you have a valid character set that is allowed here? Currently the name has no validation other than length, shall we impose some restrictions?

Reply:
Added the pattern validation.